lib/OpenQA/Worker/Cache/Request.pm

```diff
@@ -0,0 +1,69 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Worker::Cache::Request;
+use OpenQA::Worker::Cache::Request::Sync;
+use OpenQA::Worker::Cache::Request::Asset;
+use Mojo::Base -base;
+use Carp 'croak';
+use OpenQA::Worker::Cache::Client;
+
+has 'task';
+has client => sub { OpenQA::Worker::Cache::Client->new };
+
+sub asset { OpenQA::Worker::Cache::Request::Asset->new(client => shift->client, @_,) }
+sub rsync { OpenQA::Worker::Cache::Request::Sync->new(client => shift->client, @_,) }
+
+sub execute_task { $_[0]->client->execute_task(shift) }
+sub status       { $_[0]->client->status(shift) }
+sub processed    { $_[0]->client->processed(shift) }
+sub output       { $_[0]->client->output(shift) }
+sub result       { $_[0]->client->result(shift) }
+sub enqueue      { $_[0]->client->enqueue(shift) }
+
+sub lock    { croak 'Not implemented in ' . __PACKAGE__ }
+sub to_hash { croak 'Not implemented in ' . __PACKAGE__ }
```

Comment:
Maybe include the name of the abstract method that's missing from the subclass. You know how it can be with method chains. 😉